airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java

```diff
@@ -123,47 +146,44 @@ private CheckedBiConsumer<AirbyteStreamNameNamespacePair, SerializableBuffer, Ex
       }
 
       final BigQueryWriteConfig writeConfig = writeConfigs.get(pair);
+      final String datasetId = writeConfig.datasetId();
+      final String stream = writeConfig.streamName();
       try (writer) {
         writer.flush();
-        final String stagedFile = bigQueryGcsOperations.uploadRecordsToStage(writeConfig.datasetId(), writeConfig.streamName(), writer);
-        // TODO: (ryankfu) needs to remove this since we're no longer storing staged files to later upload
+        final String stagedFile = bigQueryGcsOperations.uploadRecordsToStage(datasetId, stream, writer);
+        /*
+         * The primary reason for still adding staged files despite immediately uploading the staged
+         * file to the destination's raw table is because the cleanup for the staged files will occur
+         * at the end of the sync
+         */
         writeConfig.addStagedFile(stagedFile);
+        bigQueryGcsOperations.copyIntoTargetTableFromStage(datasetId, stream, writeConfig.targetTableId(), writeConfig.tableSchema(),
+            List.of(stagedFile));
       } catch (final Exception e) {
-        LOGGER.error("Failed to flush and upload buffer to stage:", e);
-        throw new RuntimeException("Failed to upload buffer to stage", e);
+        LOGGER.error("Failed to flush and commit buffer data into destination's raw table:", e);
+        throw new RuntimeException("Failed to upload buffer to stage and commit to destination", e);
       }
     };
   }
 
+  /**
+   * Tear down process, will attempt to clean out any staging area
+   *
+   * @param bigQueryGcsOperations collection of staging operations
+   * @param writeConfigs
+   * @return
+   */
   private CheckedConsumer<Boolean, Exception> onCloseFunction(final BigQueryStagingOperations bigQueryGcsOperations,
                                                               final Map<AirbyteStreamNameNamespacePair, BigQueryWriteConfig> writeConfigs) {
     return (hasFailed) -> {
-      if (!hasFailed) {
-        LOGGER.info("Copying into tables in destination started for {} streams", writeConfigs.size());
-
-        for (final BigQueryWriteConfig writeConfig : writeConfigs.values()) {
-          final String datasetId = writeConfig.datasetId();
-          final String stream = writeConfig.streamName();
-
-          try {
-            bigQueryGcsOperations.copyIntoTmpTableFromStage(datasetId, stream, writeConfig.tmpTableId(), writeConfig.tableSchema(),
-                writeConfig.stagedFiles());
-          } catch (final Exception e) {
-            bigQueryGcsOperations.cleanUpStage(datasetId, stream, writeConfig.stagedFiles());
-            final String stagingPath = bigQueryGcsOperations.getStagingFullPath(datasetId, stream);
-            throw new RuntimeException("Failed to upload data from stage " + stagingPath, e);
-          }
-          writeConfig.clearStagedFiles();
-          bigQueryGcsOperations.copyIntoTargetTable(
```

Comment:
After these changes `copyIntoTargetTable` is not called anymore. Can we safely delete `copyIntoTargetTable` from `BigQueryStagingOperations` as well as from `BigQueryGcsOperations`? Once we do that, can we also delete `AbstractBigQueryUploader::partitionIfUnpartitioned`?

Reply:
Yes, I can remove `copyIntoTargetTable`. I was aiming to clean up this but didn't write a note down. Thanks for catching this and allowing for more trimmed code

As for `AbstractBigQueryUploader::partitionIfUnpartitioned`, I'm cannot _safely_ remove that method yet since it's currently in use for BigQueryDenormalized and since it's not clear how the code flows through that connector I'm opting to limit scope to get the Staging Checkpointing work across first

Reply:
OK, we can keep `partitionIfUnpartitioned` for now